pkg/apis/streaming/v1alpha1/processor_lifecycle.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2019 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	kedav1alpha1 "github.com/kedacore/keda/pkg/apis/keda/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/projectriff/system/pkg/apis"
+	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
+)
+
+const (
+	ProcessorConditionReady = apis.ConditionReady
+	// TODO add aggregated streams ready status
+	ProcessorConditionFunctionReady     apis.ConditionType = "FunctionReady"
+	ProcessorConditionDeploymentReady   apis.ConditionType = "DeploymentReady"
+	ProcessorConditionScaledObjectReady apis.ConditionType = "ScaledObjectReady"
+)
+
+var processorCondSet = apis.NewLivingConditionSet(
+	ProcessorConditionFunctionReady,
+	ProcessorConditionDeploymentReady,
+	ProcessorConditionScaledObjectReady,
+)
+
+func (ps *ProcessorStatus) GetObservedGeneration() int64 {
+	return ps.ObservedGeneration
+}
+
+func (ps *ProcessorStatus) IsReady() bool {
+	return processorCondSet.Manage(ps).IsHappy()
+}
+
+func (*ProcessorStatus) GetReadyConditionType() apis.ConditionType {
+	return ProcessorConditionReady
+}
+
+func (ps *ProcessorStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return processorCondSet.Manage(ps).GetCondition(t)
+}
+
+func (ps *ProcessorStatus) InitializeConditions() {
+	processorCondSet.Manage(ps).InitializeConditions()
+}
+
+func (ps *ProcessorStatus) MarkFunctionNotFound(name string) {
+	processorCondSet.Manage(ps).MarkFalse(ProcessorConditionFunctionReady, "NotFound",
+		"Unable to find function %q.", name)
+}
+
+func (ps *ProcessorStatus) PropagateFunctionStatus(fs *buildv1alpha1.FunctionStatus) {
+	ps.FunctionImage = fs.LatestImage
+	ps.FunctionImage = "ericbottard/fn" // TODO remove once build ctrl exists
+
+	if ps.FunctionImage == "" {
+		processorCondSet.Manage(ps).MarkFalse(ProcessorConditionFunctionReady, "NoImage",
+			"Function has no latestImage")
+	} else {
+		processorCondSet.Manage(ps).MarkTrue(ProcessorConditionFunctionReady)
+	}
+}
+
+func (ps *ProcessorStatus) MarkDeploymentNotOwned(name string) {
+	processorCondSet.Manage(ps).MarkFalse(ProcessorConditionDeploymentReady, "NotOwned",
+		"There is an existing Deployment %q that we do not own.", name)
+}
+
+func (ps *ProcessorStatus) MarkScaledObjectNotOwned(name string) {
```

Comment:
ProcessorStatus has a ScaledObjectName field we can use for the message

```suggestion
func (ps *ProcessorStatus) MarkScaledObjectNotOwned() {
```